kayenta-core/src/main/java/com/netflix/kayenta/canary/ExecutionMapper.java

```diff
@@ -89,104 +81,66 @@ public ExecutionMapper(
-                        "Unable to find stage '"
-                            + CanaryStageNames.REFID_JUDGE
-                            + "' in pipeline ID '"
-                            + canaryExecutionId
-                            + "'"));
+    Stage judgeStage = getStageFromExecution(pipeline, CanaryStageNames.REFID_JUDGE);
     Map<String, Object> judgeOutputs = judgeStage.getOutputs();
 
-    Stage contextStage =
-        pipeline.getStages().stream()
-            .filter(stage -> stage.getRefId().equals(CanaryStageNames.REFID_SET_CONTEXT))
-            .findFirst()
-            .orElseThrow(
-                () ->
-                    new IllegalArgumentException(
-                        "Unable to find stage '"
-                            + CanaryStageNames.REFID_SET_CONTEXT
-                            + "' in pipeline ID '"
-                            + canaryExecutionId
-                            + "'"));
+    Stage contextStage = getStageFromExecution(pipeline, CanaryStageNames.REFID_SET_CONTEXT);
     Map<String, Object> contextContext = contextStage.getContext();
 
-    Stage mixerStage =
-        pipeline.getStages().stream()
-            .filter(stage -> stage.getRefId().equals(CanaryStageNames.REFID_MIX_METRICS))
-            .findFirst()
-            .orElseThrow(
-                () ->
-                    new IllegalArgumentException(
-                        "Unable to find stage '"
-                            + CanaryStageNames.REFID_MIX_METRICS
-                            + "' in pipeline ID '"
-                            + canaryExecutionId
-                            + "'"));
-    Map<String, Object> mixerContext = mixerStage.getContext();
+    Stage mixerStage = getStageFromExecution(pipeline, CanaryStageNames.REFID_MIX_METRICS);
     Map<String, Object> mixerOutputs = mixerStage.getOutputs();
 
     CanaryExecutionStatusResponse.CanaryExecutionStatusResponseBuilder
         canaryExecutionStatusResponseBuilder =
             CanaryExecutionStatusResponse.builder()
                 .application((String) contextContext.get("application"))
                 .parentPipelineExecutionId((String) contextContext.get("parentPipelineExecutionId"))
-                .pipelineId(pipeline.getId())
-                .storageAccountName(storageAccountName);
-    if (contextContext.containsKey("canaryConfigId")) {
-      String canaryConfigId = (String) contextContext.get("canaryConfigId");
-      canaryExecutionStatusResponseBuilder.canaryConfigId(canaryConfigId);
-    }
-    if (contextContext.containsKey("configurationAccountName")) {
-      String configurationAccountName = (String) contextContext.get("configurationAccountName");
-      canaryExecutionStatusResponseBuilder.configurationAccountName(configurationAccountName);
-    }
+                .pipelineId(pipeline.getId());
+
+    pipeline.getStages().stream()
+        .filter(stage -> stage.getName().startsWith("fetch"))
```

Comment:
not super familiar with the inner workings/usage of orca here but looking for stage with a given name seems brittle... can you key off stage type here?

Reply:
Yes, I think you are correct. 

Even better I think I will update the PR to just store this piece of information in the setupContext stage rather than even needed to rely on filtering one of the fetch stages.